python/ray/autoscaler/aws/config.py

```diff
@@ -156,33 +156,25 @@ def _configure_subnet(config):
             "and trying this again. Note that the subnet must map public IPs "
             "on instance launch.")
     if "availability_zone" in config["provider"]:
-        default_subnet = next((
-            s for s in subnets
-            if s.availability_zone == config["provider"]["availability_zone"]),
-                              None)
-        if not default_subnet:
+        azs = config["provider"]["availability_zone"].split(',')
+        subnets = [s for s in subnets if s.availability_zone in azs]
+        if not subnets:
             raise Exception(
                 "No usable subnets matching availability zone {} "
                 "found. Choose a different availability zone or try "
                 "manually creating an instance in your specified region "
                 "to populate the list of subnets and trying this again."
                 .format(config["provider"]["availability_zone"]))
-    else:
-        default_subnet = subnets[0]
-
-    if "SubnetId" not in config["head_node"]:
-        assert default_subnet.map_public_ip_on_launch, \
-            "The chosen subnet must map nodes with public IPs on launch"
-        config["head_node"]["SubnetId"] = default_subnet.id
-        print("SubnetId not specified for head node, using {} in {}".format(
-            default_subnet.id, default_subnet.availability_zone))
-
-    if "SubnetId" not in config["worker_nodes"]:
-        assert default_subnet.map_public_ip_on_launch, \
-            "The chosen subnet must map nodes with public IPs on launch"
-        config["worker_nodes"]["SubnetId"] = default_subnet.id
-        print("SubnetId not specified for workers, using {} in {}".format(
-            default_subnet.id, default_subnet.availability_zone))
+
+    subnet_ids = [s.subnet_id for s in subnets]
+    subnet_descr = [(s.subnet_id, s.availability_zone) for s in subnets]
+    if "SubnetIds" not in config["head_node"]:
```

Comment:
Does it make sense to keep the map_public_ip_on_launch filter, at least for the head node? I know we don't require it for worker nodes any more.

Reply:
This check is already performed on line 149; the later checks I believe are redundant so I removed them.